frontend/src/views/WheresMyGeneV2/components/GeneSearchBar/components/SourceDataButton/index.tsx

```diff
@@ -2,17 +2,30 @@ import { MouseEventHandler } from "react";
 import { track } from "src/common/analytics";
 import { EVENTS } from "src/common/analytics/events";
 import { EXCLUDE_IN_SCREENSHOT_CLASS_NAME } from "../SaveExport";
-import { StyledButtonDiv, StyledButtonIcon, StyledLabel } from "./style";
+import {
+  BadgeCounter,
+  StyledButtonDiv,
+  StyledButtonIcon,
+  StyledLabel,
+} from "./style";
 
 export default function SourceDataButton({
   handleRightSidebarButtonClick,
+  referenceCount,
 }: {
   handleRightSidebarButtonClick: MouseEventHandler<HTMLButtonElement>;
+  referenceCount: number;
 }): JSX.Element {
   return (
     <StyledButtonDiv className={EXCLUDE_IN_SCREENSHOT_CLASS_NAME}>
       <StyledLabel>Source Data</StyledLabel>
-
+      {referenceCount > 0 && (
+        <BadgeCounter
+          badgeContent={referenceCount}
+          width={referenceCount > 99 ? "28px" : "20px"}
+          data-testid={"source-data-badge"}
```

Comment:
nit: we can save an expression here by just passing in a string without the curly brackets!
```suggestion
          data-testid="source-data-badge"
```